subsys/net/ip/net_if.c

```diff
@@ -4448,12 +4453,17 @@ static void init_igmp(struct net_if *iface)
 
 int net_if_up(struct net_if *iface)
 {
+	const struct device *dev;
 	int status = 0;
 
 	NET_DBG("iface %p", iface);
 
 	net_if_lock(iface);
 
+	/* If the device is not ready it is pointless trying to take it up. */
+	dev = net_if_get_device(iface);
+	NET_ASSERT(dev);
+
```

Comment:
Forgot to move this next to the `device_is_ready()` call, will send a new version that will do that.